lib/osut/utils.rb

```diff
@@ -6085,6 +6388,27 @@ def addSkyLights(spaces = [], opts = {})
     wl  = w0 + gap
     wl2 = wl * wl
 
+    # Validate requested skylight-to-roof ratio (or overall area).
+    if opts.key?(:area)
+      if opts[:area].respond_to?(:to_f)
+        area = opts[:area].to_f
+        log(WRN, "Area reset to 0.0m2 (#{mth})") if area < 0
+      else
+        return mismatch("area", opts[:area], Numeric, mth, DBG, 0)
+      end
+    elsif opts.key?(:srr)
```

Comment:
It is often preferable to precalculate a desired total skylight **:area** (from a building's _grossRoofArea_ **x** a code-required SRR%) than relying on the **:srr** option directly. Once **:area** is determined, users may (wisely) choose to filter out technical or ancillary spaces (e.g. washrooms, corridors) as potential candidates for skylights. The value of **:area** would remain valid - not the :**srr**.